### PR TITLE
Get rid of the dotmap and generate Munches for multsigs and extras too.

### PR DESCRIPTION
**Pull request by Tritium-VLK**

### CODE REVIEW

File: bal_addresses/addresses.py [hunk 1]
```diff
@@ -73,9 +75,30 @@ def deployments(self) -> Optional[Munch]:
...
 
+    @property
+    def extras(self) -> Optional[Munch]:
+        """
+        Get the extras for all chains in a form of a Munch object
+        """
+        if self._extras is not None:
+            return self._extras
+        else:
+            self.populate_extras()
```
Comment (SHAKOTN) [inline bal_addresses/addresses.py:88]: I am failing to find function implementation here?

File: bal_addresses/addresses.py [hunk 2]
```diff
@@ -73,9 +75,30 @@ def deployments(self) -> Optional[Munch]:
...
+
+    @property
+    def multisigs(self) -> Optional[Munch]:
+        """
+        Get the multisigs for all chains in a form of a Munch object
+        """
+        if self._multisigs is not None:
+            return self._multisigs
+        else:
+            self.populate_multisigs()
```
Comment (SHAKOTN) [inline bal_addresses/addresses.py:99]: Same here

File: bal_addresses/addresses.py (hunk 1 above)
Comment (SHAKOTN) [reply bal_addresses/addresses.py:88]: This will literally revert with error if you try to access the property

**Review by SHAKOTN — CHANGES_REQUESTED** [review]
I don't see implementation of `self.populate_extras()` and `self.populate_multisigs()`. Accessing property will literally throw an exception

**Comment by Tritium-VLK** [comment]
ok @SHAKOTN please take another look.

File: bal_addresses/addresses.py [hunk 3]
```diff
@@ -104,41 +124,73 @@ def _process_deployment(self, deployment: Dict) -> Dict:
...
+
+    def populate_multisigs(self) -> None:
+        msigs = requests.get(
+            f"{GITHUB_RAW_EXTRAS}/multisigs.json"
+        ).json()
+        if msigs.get(self.chain):
+            self._multisigs = Munch.fromDict(msigs[self.chain])
+        else:
+            print(f"Warning: No multisigs for chain {self.chain}, multisigs must be added in extras/multisig.json")
+            self._multisigs = {}
```
Comment (SHAKOTN) [inline bal_addresses/addresses.py:144]: Make it empty Munch object to unify the interface

File: bal_addresses/errors.py [hunk 4]
```diff
@@ -0,0 +1,8 @@
+
+class Errors():
```
Comment (SHAKOTN) [inline bal_addresses/errors.py:2]: Not very pythonic. You just declare errors outside Errors class
```suggestion
    class MultipleMatchesError(Exception):
        pass

    class NoResultError(Exception):
        pass
```

File: tests/test_addresses.py [hunk 5]
```diff
@@ -76,10 +105,36 @@ def test_deployments_invalid_format():
             }
         }
     )
+    responses.add(
```
Comment (SHAKOTN) [inline tests/test_addresses.py:108]: Nicely done, but please pay attention to the name of the test.

It's always good to make tests as small as possible.

My suggestion: create separate test functions for both extras and msigs logic

File: bal_addresses/errors.py (hunk 4 above)
Comment (Tritium-VLK) [reply bal_addresses/errors.py:2]: The thing is that the errors often need to be caught as part of the usage pattern, and it's kind of a pain to have to specifiy each error/track them all at the top level/not in a class.

Is there a better way to handle this.

https://github.com/BalancerMaxis/multisig-ops/blob/989d99d459ef2bbc687d8150e2e29d61f8a717d4/action-scripts/gen_add_permissions_payload.py#L69

Should the error be inside the AddrBook and BalPermissions classeS?  Then have the same error twice.

File: tests/test_addresses.py (hunk 5 above)
Comment (Tritium-VLK) [reply tests/test_addresses.py:108]: I tried that but you have to copy all the inputs each time and then there is very little additional code.  It felt silly and made changes to the test inputs  hard.

Comment (SHAKOTN) [reply tests/test_addresses.py:108]: You should separate all the test cases in smaller units, that is my recommendation. It's better to NOT test everything in one function even if it leads to boilerplate code

File: bal_addresses/errors.py (hunk 4 above)
Comment (SHAKOTN) [reply bal_addresses/errors.py:2]: I don't get what you mean. Why not just:
```python
from from bal_addresses.errors import NoResultError
...
except NoResultError as err:
```
???

File: tests/test_addresses.py (hunk 5 above)
Comment (Tritium-VLK) [reply tests/test_addresses.py:108]: Another day ser :)

**Review by SHAKOTN — APPROVED** [review]

**Pull request merged**